pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -0,0 +1,183 @@
+
+// Object returns the job instance.
+func (j *Pod) Object() client.Object {
+	return (*corev1.Pod)(j)
+}
+
+func (p *Pod) gateIndex() int {
+
+	for i := range p.Spec.SchedulingGates {
+		if p.Spec.SchedulingGates[i].Name == ScheduleingGateName {
+			return i
+		}
+	}
+	return gateNotFound
+}
+
+// IsSuspended returns whether the job is suspended or not.
+func (p *Pod) IsSuspended() bool {
+	return p.gateIndex() != gateNotFound
+}
+
+// Suspend will suspend the job.
+func (p *Pod) Suspend() {
+	// TODO: maybe change the framework so this can provide feedback,
+	// the pod can only be "suspended" by the mutation hook if it's not
+	// done the only way to potentialy stop its execution is the eviction
+	// which will also terminate the pod.
+}
+
+// RunWithPodSetsInfo will inject the node affinity and podSet counts extracting from workload to job and unsuspend it.
+func (p *Pod) RunWithPodSetsInfo(podSetsInfo []jobframework.PodSetInfo) {
+	if len(podSetsInfo) != 1 {
+		return
+	}
+	idx := p.gateIndex()
+	if idx != gateNotFound {
+		p.Spec.SchedulingGates = append(p.Spec.SchedulingGates[:idx], p.Spec.SchedulingGates[idx+1:]...)
+	}
+
+	// TODO: manage the node selector
+	// NOTE: it's only possible to add and only if k8s > 1.27 is used, case in which, since if the provided
+	// selectors are changing a existing key will fail we should be able to "refuse" the assignment
+	// p.Spec.NodeSelector = maps.MergeKeepFirst(podSetsInfo[0].NodeSelector, p.Spec.NodeSelector)
+}
+
+// RestorePodSetsInfo will restore the original node affinity and podSet counts of the job.
+func (p *Pod) RestorePodSetsInfo(nodeSelectors []jobframework.PodSetInfo) {
+	// TODO: find a way to evict the pod from its node
+}
+
+// Finished means whether the job is completed/failed or not,
+// condition represents the workload finished condition.
+func (p *Pod) Finished() (metav1.Condition, bool) {
+
+	ph := p.Status.Phase
+	condition := metav1.Condition{
+		Type:    kueue.WorkloadFinished,
+		Status:  metav1.ConditionTrue,
+		Reason:  "JobFinished",
+		Message: "Job finished successfully",
+	}
+	if ph == corev1.PodFailed {
+		condition.Message = "Job failed"
+	}
+
+	return condition, ph == corev1.PodSucceeded || ph == corev1.PodFailed
+}
+
+// PodSets will build workload podSets corresponding to the job.
+func (p *Pod) PodSets() []kueue.PodSet {
+	return []kueue.PodSet{
+		{
+			Name:  kueue.DefaultPodSetName,
+			Count: 1,
+			Template: corev1.PodTemplateSpec{
+				Spec: *p.Spec.DeepCopy(),
+			},
+		},
+	}
+}
+
+// PriorityClass returns the job's priority class name.
+func (p *Pod) PriorityClass() string {
+	return p.Spec.PriorityClassName
+}
+
+// IsActive returns true if there are any running pods.
+func (p *Pod) IsActive() bool {
+	return p.Status.Phase == corev1.PodRunning
+}
+
+// PodsReady instructs whether job derived pods are all ready now.
+func (p *Pod) PodsReady() bool {
+	ph := p.Status.Phase
+	return ph == corev1.PodRunning || ph == corev1.PodSucceeded
```

Comment:
Actually, as the name implies, we should be checking for the `Ready` condition.

Not sure if the phase Succeeded matters, given that at that point the Workload is considered Finished. Is this function called in this scenario? Same would happen with Failed.